src/frontend/components/DashboardVideoPaneDownloadOption/index.spec.tsx

```diff
@@ -0,0 +1,111 @@
+import { cleanup, fireEvent, render, wait } from '@testing-library/react';
+import fetchMock from 'fetch-mock';
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { Provider } from 'react-redux';
+
+import { DashboardVideoPaneDownloadOption } from '.';
+import { bootstrapStore } from '../../data/bootstrapStore';
+import { appState } from '../../types/AppData';
+import { uploadState } from '../../types/tracks';
+
+describe('<DashboardVideoPaneDownloadOption />', () => {
+  afterEach(cleanup);
+  afterEach(fetchMock.restore);
+
+  const video = {
+    description: 'Some description',
+    id: '442',
+    is_ready_to_play: true,
+    show_download: false,
+    thumbnail: null,
+    timed_text_tracks: [],
+    title: 'Some title',
+    upload_state: uploadState.READY,
+    urls: {
+      manifests: {
+        dash: 'https://example.com/dash.mpd',
+        hls: 'https://example.com/hls.m3u8',
+      },
+      mp4: {
+        144: 'https://example.com/144p.mp4',
+        240: 'https://example.com/240p.mp4',
+        480: 'https://example.com/480p.mp4',
+        720: 'https://example.com/720p.mp4',
+        1080: 'https://example.com/1080p.mp4',
+      },
+      thumbnails: {
+        144: 'https://example.com/144p.jpg',
+        240: 'https://example.com/240p.jpg',
+        480: 'https://example.com/480p.jpg',
+        720: 'https://example.com/720p.jpg',
+        1080: 'https://example.com/1080p.jpg',
+      },
+    },
+  };
+
+  it('renders with checkbox not checked', () => {
+    const { getByLabelText } = render(
+      <Provider
+        store={bootstrapStore({
+          jwt: '',
+          resourceLinkid: '',
+          state: appState.INSTRUCTOR,
+          video,
+        })}
+      >
+        {' '}
+        <DashboardVideoPaneDownloadOption video={video} />
+      </Provider>,
+    );
+
+    expect(getByLabelText('Allow video download')).toHaveProperty(
+      'checked',
+      false,
+    );
+  });
+
+  it('updates the checkbox and the video record when the user clicks the checkbox', async () => {
+    fetchMock.mock(
```

Comment:
IMO here again you are doing it wrong by testing the fetch call. You should only mock the `updateResource` and test if it has been called. Nothing else.

Reply:
Discuss about this this morning, We keep it like this.